pkg/controllermanager/controller/bastion/reconciler.go

```diff
@@ -42,6 +43,9 @@ type Reconciler struct {
 func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log := logf.FromContext(ctx)
 
+	ctx, cancel := context.WithTimeout(ctx, controllerutils.DefaultReconciliationTimeout)
```

Comment:
Yes, I wondered the same thing, sadly I could not find any way to do it. The context with which the manager is started is directly passed to the reconcile functions without being modified anywhere.

Reply:
Hmm, it is definitely modified here: https://github.com/kubernetes-sigs/controller-runtime/blob/3da2de04734a0ef63d2e6e704eea973b327ab1ac/pkg/internal/controller/controller.go#L318-L319
This might be a good place to inject the timeout as well.

Reply:
Do you mean to modify the controller-runtime library itself? If so than yes, a function option could be added that modifies the context and adds a timeout to it in that place. I also thought about that but wondered if I should propose it. There is already this reply: https://github.com/kubernetes-sigs/controller-runtime/issues/798#issuecomment-707197729 which I don't quite understand if it is against such timeouts. 
And back when the reconciler functions did not accept a context parameter there was this issue https://github.com/kubernetes-sigs/controller-runtime/issues/265 which has a similar idea about adding a ContextBuilder that can be used to create a context with timeouts.